### ISSUE TITLE
Retry 5xx errors

### ISSUE THREAD

**Issue by EnricoMi** [EnricoMi/download-buildkite-artifact-action#20]
Just saw this error while waiting for job to finish:
```
Thu, 21 Apr 2022 12:56:52 GMT 2022-04-21 12:56:52 +0000 - download-buildkite-artifact -  INFO - Waiting for build 7596 to finish.
Thu, 21 Apr 2022 12:56:52 GMT 2022-04-21 12:56:52 +0000 - download-buildkite-artifact -  INFO - Build is in scheduled state.
Thu, 21 Apr 2022 12:57:23 GMT 2022-04-21 12:57:23 +0000 - download-buildkite-artifact -  INFO - Build is in running state.
Thu, 21 Apr 2022 18:40:00 GMT Traceback (most recent call last):
Thu, 21 Apr 2022 18:40:00 GMT   File "/action/download_artifacts.py", line 416, in <module>
Thu, 21 Apr 2022 18:40:00 GMT     commit, output_path):
Thu, 21 Apr 2022 18:40:00 GMT   File "/action/download_artifacts.py", line 277, in main
Thu, 21 Apr 2022 18:40:00 GMT     build = get_build(buildkite, org, pipeline, build_number)
Thu, 21 Apr 2022 18:40:00 GMT   File "/action/download_artifacts.py", line 64, in get_build
Thu, 21 Apr 2022 18:40:00 GMT     return buildkite.builds().get_build_by_number(org, pipeline, build_number, include_retried_jobs=True)
Thu, 21 Apr 2022 18:40:00 GMT   File "/usr/local/lib/python3.6/site-packages/pybuildkite/builds.py", line 255, in get_build_by_number
Thu, 21 Apr 2022 18:40:00 GMT     query_params=query_params,
Thu, 21 Apr 2022 18:40:00 GMT   File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 131, in get
Thu, 21 Apr 2022 18:40:00 GMT     as_stream=as_stream,
Thu, 21 Apr 2022 18:40:00 GMT   File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 82, in request
Thu, 21 Apr 2022 18:40:00 GMT     response.raise_for_status()
Thu, 21 Apr 2022 18:40:00 GMT   File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 953, in raise_for_status
Thu, 21 Apr 2022 18:40:00 GMT     raise HTTPError(http_error_msg, response=self)
Thu, 21 Apr 2022 18:40:00 GMT requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://api.buildkite.com/v2/organizations/horovod/pipelines/horovod/builds/7596?include_retried_jobs=True&per_page=100
```

**Comment by EnricoMi** [comment]
Fixed in bff61b0.

**Comment by EnricoMi** [comment]
Fix works:

```
2022-04-22 17:50:17 +0000 - download-buildkite-artifact -  INFO - Waiting for build 7617 to finish.
2022-04-22 17:50:17 +0000 - download-buildkite-artifact -  INFO - Build is in scheduled state.
2022-04-22 17:51:49 +0000 - download-buildkite-artifact -  INFO - Build is in running state.
2022-04-22 19:05:27 +0000 - download-buildkite-artifact -  INFO - Getting build 7617 failed, retrying in 30s.
Traceback (most recent call last):
  File "/action/download_artifacts.py", line 279, in main
    build = get_build(buildkite, org, pipeline, build_number)
  File "/action/download_artifacts.py", line 64, in get_build
    return buildkite.builds().get_build_by_number(org, pipeline, build_number, include_retried_jobs=True)
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/builds.py", line 255, in get_build_by_number
    query_params=query_params,
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 131, in get
    as_stream=as_stream,
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 82, in request
    response.raise_for_status()
  File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://api.buildkite.com/v2/organizations/horovod/pipelines/horovod/builds/7617?include_retried_jobs=True&per_page=100
2022-04-22 19:34:35 +0000 - download-buildkite-artifact -  INFO - Getting build 7617 failed, retrying in 30s.
Traceback (most recent call last):
  File "/action/download_artifacts.py", line 279, in main
    build = get_build(buildkite, org, pipeline, build_number)
  File "/action/download_artifacts.py", line 64, in get_build
    return buildkite.builds().get_build_by_number(org, pipeline, build_number, include_retried_jobs=True)
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/builds.py", line 255, in get_build_by_number
    query_params=query_params,
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 131, in get
    as_stream=as_stream,
  File "/usr/local/lib/python3.6/site-packages/pybuildkite/client.py", line 82, in request
    response.raise_for_status()
  File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://api.buildkite.com/v2/organizations/horovod/pipelines/horovod/builds/7617?include_retried_jobs=True&per_page=100
2022-04-22 19:43:53 +0000 - download-buildkite-artifact -  INFO - Build is in passed state.
```
